apps/openmw/mwgui/tooltips.cpp

```diff
@@ -351,35 +353,40 @@ namespace MWGui
             ToolTipInfo info = object.getToolTipInfo(mFocusObject, count);
             if (!image)
                 info.icon = "";
-            tooltipSize = createToolTip(info, true);
+            tooltipSize = createToolTip(info, true, isOwned);
         }
 
         return tooltipSize;
     }
     
     bool ToolTips::checkOwned()
     {
-        if(!mFocusObject.isEmpty())
-        {
-            MWWorld::Ptr ptr = MWMechanics::getPlayer();
-            MWWorld::Ptr victim;
-            
-            MWBase::MechanicsManager* mm = MWBase::Environment::get().getMechanicsManager();
-            bool allowed = mm->isAllowedToUse(ptr, mFocusObject, victim); 
+        if(mFocusObject.isEmpty())
+            return false;
 
-            return !allowed;
-        }
-        else
+        MWWorld::Ptr ptr = MWMechanics::getPlayer();
+        MWWorld::Ptr victim;
+        MWWorld::Ptr target = mFocusObject;
+
+        // if a focused object is stored in container, check if the player can take items from this container instead if owner check
+        if (mFocusObject.getContainerStore() &&
+            MWBase::Environment::get().getWindowManager()->getMode() == GM_Container)
         {
-            return false;
+            // just make sure that focused object is stored in the current opened container
+            MWWorld::Ptr container = MWBase::Environment::get().getWindowManager()->getContainerWindow()->getPtr();
+            if (!container.isEmpty() && mFocusObject.getContainerStore() == &container.getClass().getContainerStore(container))
```

Comment:
This solution is a bit unclean, but at least we do not have to iterate over all containers in all active cells.